ocaml/overrides.js

```diff
@@ -1,6 +1,17 @@
-/* global globalThis
+/* global caml_named_value, caml_global_data, caml_string_of_jsstring
  */
 
+//Provides: caml_wrap_exception const (const)
+//Requires: caml_global_data,caml_string_of_jsstring,caml_named_value
+//Requires: caml_return_exn_constant
+function caml_wrap_exception(e) {
+  if (e instanceof Array) return e;
+  if (e instanceof globalThis.Error && caml_named_value('jsError'))
+    return [0, caml_named_value('jsError'), e];
+  //fallback: wrapped in Failure
+  return [0, caml_global_data.Failure, caml_string_of_jsstring(String(e))];
+}
```

Comment:
this fixes that stack overflows used to be thrown as an array. it overrides the `caml_wrap_exception` function from JSOO, which has a special case for stack overflows, which doesn't presever the stack trace.

this simply removes the special case, and thus stack overflows end up in the first `if` case, where we preserve the original error `e` which can be later be re-thrown in JS.